### ISSUE TITLE
Gathering stats

### ISSUE THREAD

**Issue by cristaloleg** [allegro/bigcache#55]
I've a proposal to collect hit and miss(something else?) metrics.
So it'll allow to see performance of the cache and tweak TLL and size of the data.

If this will be accepted I would like to be assigned to this 😉

**Comment by janisz** [comment]
I'd love to merge it!

We need to answer following questions:
- Are we going to collect global hit and miss or per each key?
- Where do we store the data? 
- How do we want to expose this metric?

I'd prefer to not use any external libs in this project. Hits&Miss could be implemented as a simple counter.

**Comment by cristaloleg** [comment]
Nononono, no external stuff :)

1. I think per-key will be too heavy, at least for 1st implementation.
2. Per shard in map, I'm not sure about `sync.Map` 'cause collecting stats is more write oriented, but `sync.Map` is more for read oriented.
3. WDYT about `cache.Stats() map[string]int64` ? and keys of this map are `hits`, `misses`, `avgShardSize`(just an example)

**Comment by cristaloleg** [comment]
2. I forgot about `sync.Atomic`, all our counters are int64 which is quite obvious. So no locks or smth for stats, and we'll create a map only on `cache.Stats()` which is great.

**Comment by janisz** [comment]
1. I concur. For first step metric per shard should be enough but we will expose them in aggregated form.
2. So we need to add stats struct to cache shard and update values with `sync.Atomic`
3. How about formalized structure instead of map? e,g.: For the first step, I'll stick with Hits and Misses only.

    ``` go
    type Stats struct {
        Hits uint64
        Misses uint64        
    }
    ```

**Comment by cristaloleg** [comment]
3/3  👍 soon I'll push changes.

**Comment by janisz** [comment]
Fixed by #57
